src/components/feedback-particle-effect/feedback-text-effects.ts

```diff
@@ -1,159 +1,85 @@
-        );
-      }
-    });
-    this.convertToParticle();
-  }
 
-  private handleTextWorkerMessage(event: MessageEvent): void {
-    this.particles = event.data.map(
-      ({ x, y, color }) => new TextParticle(this, x, y, color)
-    );
+    this.loadFont();
   }
 
-  private convertToParticle(): void {
-    const imageData = this.context.getImageData(
-      0,
-      0,
-      this.canvasWidth,
-      this.canvasHeight
-    );
-    this.context.clearRect(0, 0, this.canvasWidth, this.canvasHeight);
-    this.textWorker.postMessage({
-      canvasWidth: this.canvasWidth,
-      canvasHeight: this.canvasHeight,
-      gap: this.gap,
-      pixels: imageData.data,
-    });
-  }
+  /**
+   * Loads the appropriate font based on the current language.
+   */
+  private async loadFont() {
+    const fontName = feedbackCustomFonts[lang] || feedbackTextDefault; // Determine the final font to use
+    const fontPath = `${FONT_BASE_PATH}${fontName}.ttf`; // Construct the font path
 
-  public render(): void {
-    this.context.clearRect(0, 0, this.canvasWidth, this.canvasHeight);
-    this.particles.forEach((particle) => {
-      particle.draw();
-      particle.update();
-    });
+    this.applyFontToElement(fontName, fontPath);
   }
 
-  public updateParticles(): void {
-    this.textWorker.postMessage({
-      particles: this.particles,
-      particleDuration: this.particleDuration,
-      startTime: this.startTime,
-    });
-  }
+  /**
+   * Applies the specified font to the feedback text element and adds the necessary @font-face rule.
+   * @param fontName - The name of the font to apply.
+   * @param fontPath - The path to the font file.
+   */
+  private applyFontToElement(fontName: string, fontPath: string) {
+    if (fontPath) {
+      const style = document.createElement("style");
+      style.textContent = `
+        @font-face {
+          font-family: '${fontName}';
+          src: url('${fontPath}') format('truetype');
+        }
+      `;
+      document.head.appendChild(style);
+    }
 
-  public clearParticle(): void {
-    this.particles = [];
+    if (this.feedbackTextElement) {
+      this.feedbackTextElement.style.fontFamily = `${fontName}, sans-serif`;
+    }
   }
 
-  public unregisterEventListener(): void {
-    this.textWorker.removeEventListener(
-      "message",
-      this.handleTextWorkerMessage
-    );
-    this.textWorker.terminate();
+  /**
+   * Updates the feedback text with an animation.
+   * @param text - The text to display in the feedback element.
+   */
+  public wrapText(text: string): void {
+    if (!this.feedbackTextElement) {
+      console.error("Feedback text element not found!");
+      return;
+    }
+
+    // Clear any existing timeout to prevent memory leaks
+    if (this.hideTimeoutId) {
+      clearTimeout(this.hideTimeoutId);
+      this.hideTimeoutId = null;
+    }
+
+    this.feedbackTextElement.textContent = text;
+    hideShowElement(false, this.feedbackTextElement);
```

Comment:
Question, why are we setting this to true and turning it to true 4 seconds later?

Reply:
this one is to hide the element 4 seconds later. setting true hides the code https://github.com/curiouslearning/FeedTheMonsterJS/blob/0c4e93f420545a76c1c724a177ac58a409168feb/src/common/utils.ts#L172-L183

Reply:
Ok, I think you need to separate the clearing part. It would be much cleaner to separate the setTime to a another method.

Keep this method for setting the text and call the clearing method afterwards. Lets promote separation of concerns as much as we can.

Reply:
applied please check @bernhardccodev